pkg/ccl/backupccl/restore_planning.go

```diff
@@ -1472,6 +1472,9 @@ func doRestorePlan(
 	var mainBackupManifests []BackupManifest
 	var localityInfo []jobspb.RestoreDetails_BackupLocalityInfo
 	var memReserved int64
+	defer func() {
```

Comment:
Did you mean to do this? I think memReserved is just 0 here, right?

Reply:
note that the new diff still moves this slightly.

Reply:
I think I was actually wrong here. While calling `defer mem.Shrink(memReserved)` would in fact make a copy of (0-valued) `memReserved`, by wrapping it in `defer func() {...}` we avoid making copies of the variables. See e.g. https://play.golang.com/p/pKXVmgKlrFC, linked from https://www.reddit.com/r/golang/comments/k1s78r/defer_versus_defer_func/